src/traces/parcoords/plot.js

```diff
@@ -0,0 +1,51 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var parcoords = require('./parcoords');
+var Lib = require('../../lib');
+
+module.exports = function plot(gd, cdparcoords) {
+
+    var fullLayout = gd._fullLayout;
+    var svg = fullLayout._paper;
+    var root = fullLayout._paperdiv;
+    var data = cdparcoords.map(function(d, i) {
+        var item = Lib.extendDeep(d[0]);
+        item._gdDataItem = gd.data[i];
+        return item;
+    });
+
+    var filterChanged = function() {
```

Comment:
`parcoords.js` no longer gets the `gd` object and doesn't mutate any part of it, all event callbacks are cleanly present in the short `plot.js` file.